subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/logging/GroupedOutputFixture.java

```diff
@@ -79,18 +79,19 @@ private String parse(String output) {
         tasks = new HashMap<String, GroupedTaskFixture>();
 
         String strippedOutput = stripAnsiCodes(stripWorkInProgressArea(output));
-        Matcher matcher = Pattern.compile(TASK_OUTPUT_PATTERN).matcher(strippedOutput);
+        Matcher matcher = TASK_OUTPUT_PATTERN.matcher(strippedOutput);
         while (matcher.find()) {
             String taskName = matcher.group(1);
-            String taskOutput = matcher.group(2);
+            String taskStatus = matcher.group(2);
+            String taskOutput = matcher.group(3);
             taskOutput = taskOutput.trim();
-            if (tasks.containsKey(taskName)) {
-                tasks.get(taskName).addOutput(taskOutput);
-            } else {
-                GroupedTaskFixture task = new GroupedTaskFixture(taskName);
-                task.addOutput(taskOutput);
+
+            GroupedTaskFixture task = tasks.get(taskName);
+            if (task == null) {
+                task = new GroupedTaskFixture(taskName);
                 tasks.put(taskName, task);
             }
+            task.addGroup(taskStatus, taskOutput);
```

Comment:
Although they go together, I think we should use the previous `addOutput` method and have a getter/setter for the status. To align better with TestKit, we should use [the term `outcome` for the status](https://blog.gradle.org/introducing-testkit). There will always be only one valid outcome per task regardless of how many output group there is.

Reply:
Done.